vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts

```diff
@@ -174,133 +176,133 @@ export class InstallTrackerSingleton
         this.inProgressInstalls.delete(resolvedInstall);
     }
 
-    public async canUninstall(isFinishedInstall : boolean, dotnetInstall : DotnetInstall, allowUninstallUserOnlyInstall = false) : Promise<boolean>
+    public async canUninstall(isFinishedInstall: boolean, dotnetInstall: DotnetInstall, allowUninstallUserOnlyInstall = false): Promise<boolean>
     {
-        return this.executeWithLock( false, this.installedVersionsId, async (id: string, install: DotnetInstall) =>
+        return this.executeWithLock(false, this.installedVersionsId, async (id: string, install: DotnetInstall) =>
         {
             this.eventStream.post(new RemovingVersionFromExtensionState(`Removing ${JSON.stringify(install)} with id ${id} from the state.`));
             const existingInstalls = await this.getExistingInstalls(id === this.installedVersionsId, true);
             const installRecord = existingInstalls.filter(x => IsEquivalentInstallation(x.dotnetInstall, install));
 
-            return installRecord.length === 0 || installRecord[0]?.installingExtensions?.length === 0 ||
+            return (installRecord?.length ?? 0) === 0 || installRecord[0]?.installingExtensions?.length === 0 ||
                 (allowUninstallUserOnlyInstall && installRecord[0]?.installingExtensions?.length === 1 && installRecord[0]?.installingExtensions?.includes('user'));
         }, isFinishedInstall ? this.installedVersionsId : this.installingVersionsId, dotnetInstall);
     }
 
-    public async uninstallAllRecords() : Promise<void>
+    public async uninstallAllRecords(): Promise<void>
     {
-        await this.executeWithLock( false, this.installingVersionsId, async () =>
+        await this.executeWithLock(false, this.installingVersionsId, async () =>
         {
             // This does not uninstall global things yet, so don't remove their ids.
             const installingVersions = await this.getExistingInstalls(false, true);
             const remainingInstallingVersions = installingVersions.filter(x => x.dotnetInstall.isGlobal);
             await this.extensionState.update(this.installingVersionsId, remainingInstallingVersions);
-        }, );
+        },);
```

Comment:
nit Is this normal to have comma-nothing?

Reply:
It is normal, mostly a personal preference IMO, it doesn't do anything. Good question